### ISSUE TITLE
Module "simple_bucket" issue

### ISSUE THREAD

**Issue by sreekanth3107** [terraform-google-modules/terraform-google-cloud-storage#322]
### TL;DR

Terraform configuration is failing to download `cloud-storage` with version 6 

### Expected behavior

Module need to download successfully 

### Observed behavior

Downloading registry.terraform.io/terraform-google-modules/cloud-storage/google 6.0.0 for code_bucket
app_code_bucket in .terraform/modules/code_bucket/modules/simple_bucket
global in ../../../modules/global
There are some problems with the configuration, described below.
The Terraform configuration must be valid before initialization so that
Error: Optional object type attributes are experimental
on .terraform/modules/code_bucket/modules/simple_bucket/variables.tf line 154:
154: variable "soft_delete_policy" {
This feature is currently an opt-in experiment, subject to change in future releases based on feedback.
Activate the feature for this module by adding module_variable_optional_attrs to the list of active experiments.

### Terraform Configuration

```hcl
module "code_bucket" {
  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
  version = "6.0.0"

  name          = local.bucket_name
  project_id    = local.project_id
  location      = local.resource_region
  iam_members   = local.bucket_members
}
```

### Terraform Version

```sh
1.8.5
```
